alembic/versions/82f5a89fbe59_slack_trigger_api.py

```diff
@@ -0,0 +1,39 @@
+"""slack trigger api
+
+Revision ID: 82f5a89fbe59
+Revises: d61a8bea7313
+Create Date: 2022-01-26 21:06:14.272254
+
+"""
+from alembic import op
+
+
+# revision identifiers, used by Alembic.
+revision = '82f5a89fbe59'
+down_revision = 'd61a8bea7313'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # add SLACKAPI to followup_apis
+    # with op.get_context().autocommit_block():
+    #     op.execute("ALTER TYPE followup_apis ADD VALUE IF NOT EXISTS 'SLACKAPI'")
+
+    op.execute(
+        """
+alter type "public"."followup_apis" rename to "followup_apis__old_version_to_be_dropped";
+
+create type "public"."followup_apis" as enum ('KAITAPI', 'SEDMAPI', 'SEDMV2API', 'IOOAPI', 'IOIAPI', 'SPRATAPI', 'SINISTROAPI', 'SPECTRALAPI', 'FLOYDSAPI', 'MUSCATAPI', 'SLACKAPI', 'ZTFAPI');
```

Comment:
My guess here:

The reason this works and not the one above is because you need to do a casting operation of sorts, since the ENUM order changes (note that `'SLACKAPI'` does not get appended, but ends up second from last).

What happens then is that a new replacement enum is created.  This new enum is an entirely different enum, so the database needs to be updated to cast between the old and the new one.  That is what the `using api_classname::text::"public"."followup_apis"` part does—it says change the `api_classname` column to use the new enum and use the values currently in that column to look them up *in the new enum* (the underlying numbers stored on disk will change).

This is probably a somewhat costly operation, but it doesn't seem to have caused problems for us so far.

In the future, the cheapest way to update an enum would be to only *append* values, and then use `ADD VALUE` as Michael did before.

Here's an example, borrowing from the postgres docs, illustrating the above:

```
skyportal=> CREATE TYPE mood AS ENUM ('sad', 'ok', 'happy');
CREATE TYPE
skyportal=> CREATE TABLE person (
    name text,
    current_mood mood
);
CREATE TABLE
skyportal=> INSERT INTO person VALUES ('Moe', 'happy');
INSERT 0 1
skyportal=> SELECT * FROM person WHERE current_mood = 'happy';
 name | current_mood 
------+--------------
 Moe  | happy
(1 row)

skyportal=> alter type mood rename to mood_old;
ALTER TYPE
skyportal=> create type mood as enum ('sad', 'ok', 'not happy', 'happy');
CREATE TYPE
skyportal=> alter table person alter column current_mood type mood;
ERROR:  column "current_mood" cannot be cast automatically to type mood
HINT:  You might need to specify "USING current_mood::mood".
                                                             ^
skyportal=> alter table person alter column current_mood type mood using current_mood::text::mood;
ALTER TABLE
skyportal=> SELECT * FROM person WHERE current_mood = 'happy';
 name | current_mood 
------+--------------
 Moe  | happy
(1 row)

skyportal=> alter type mood add value 'grumpy';
ALTER TYPE
skyportal=> SELECT * FROM person WHERE current_mood = 'happy';
 name | current_mood 
------+--------------
 Moe  | happy
(1 row)
```